components/camel-undertow/src/main/java/org/apache/camel/component/undertow/DefaultUndertowHttpBinding.java

```diff
@@ -325,9 +325,7 @@ public Object toHttpResponse(HttpServerExchange httpExchange, Message message) t
                 // we failed due an exception so print it as plain text
                 StringWriter sw = new StringWriter();
                 PrintWriter pw = new PrintWriter(sw);
-                ObjectHelper.notNull(exception.getCause(), "exception cause");
-
-                exception.getCause().printStackTrace(pw);
+                ObjectHelper.applyIfNotEmpty(exception.getCause(), value -> value, () -> exception).printStackTrace(pw);
```

Comment:
Can we maybe avoid abusing streaming api, and use a != null check. Also I wonder if we should not print the full stack-trace, maybe it should just be `exception.printStackTrace(pw)` instead. I don't recall a reason why we do a getCause. We likely have similar problem in some of the other http components that has this kind of binding code as well.